lib/utilities/monitoringHandler.js

```diff
@@ -1,6 +1,107 @@
 
+const collectDefaultMetrics = client.collectDefaultMetrics;
+let crrStatsPulled = false;
+const numberOfBuckets = new client.Gauge({
+    name: 'cloud_server_number_of_buckets',
+    help: 'Total number of buckets',
+});
+const numberOfObjects = new client.Gauge({
+    name: 'cloud_server_number_of_objects',
+    help: 'Total number of objects',
+});
+const dataDiskAvailable = new client.Gauge({
+    name: 'cloud_server_data_disk_available',
+    help: 'Available data disk storage in bytes',
+});
+const dataDiskFree = new client.Gauge({
+    name: 'cloud_server_data_disk_free',
+    help: 'Free data disk storage in bytes',
+});
+const dataDiskTotal = new client.Gauge({
+    name: 'cloud_server_data_disk_total',
+    help: 'Total data disk storage in bytes',
+});
+
+const labelNames = ['method', 'service', 'code'];
+const httpRequestsTotal = new client.Counter({
+    labelNames,
+    name: 'cloud_server_http_requests_total',
+    help: 'Total number of HTTP requests',
+});
+
+const httpRequestSizeBytes = new client.Summary({
+    labelNames,
+    name: 'cloud_server_http_request_size_bytes',
+    help: 'The HTTP request sizes in bytes.',
+});
+
+const httpResponseSizeBytes = new client.Summary({
+    labelNames,
+    name: 'cloud_server_http_response_size_bytes',
+    help: 'The HTTP response sizes in bytes.',
+});
+
+function promMetrics(requestType, bucketName, code, typeOfRequest,
+    newByteLength, oldByteLength, isVersionedObj,
+    numOfObjectsRemoved) {
+    httpRequestsTotal.labels(requestType, 'cloud_server', code).inc();
+    if ((typeOfRequest === 'objectPut' ||
+    typeOfRequest === 'copyObject' ||
+    typeOfRequest === 'objectPutPart') &&
+    code === '200') {
+        httpRequestSizeBytes
+            .labels(requestType, 'cloud_server', code)
+            .observe(newByteLength - (isVersionedObj ? 0 : oldByteLength));
+        dataDiskAvailable.dec(newByteLength -
+            (isVersionedObj ? 0 : oldByteLength));
+        dataDiskFree.dec(newByteLength -
+            (isVersionedObj ? 0 : oldByteLength));
+        numberOfObjects.inc();
+    }
+    if (typeOfRequest === 'createBucket' && code === '200') {
+        numberOfBuckets.inc();
+    }
+    if (typeOfRequest === 'getObject' && code === '200') {
+        httpResponseSizeBytes
+            .labels(requestType, 'cloud_server', code)
+            .observe(newByteLength);
+    }
+    if ((typeOfRequest === 'deleteBucket' ||
+    typeOfRequest === 'deleteBucketWebsite')
+    && code === '200') {
+        numberOfBuckets.dec();
+    }
+    if ((typeOfRequest === 'deleteObject' ||
+    typeOfRequest === 'abortMultipartUpload' ||
+    typeOfRequest === 'multiObjectDelete')
+    && code === '200') {
+        dataDiskAvailable.inc(newByteLength);
+        dataDiskFree.inc(newByteLength);
+        if (numOfObjectsRemoved) {
+            numberOfObjects.dec(numOfObjectsRemoved);
+        } else {
+            numberOfObjects.dec();
+        }
+    }
+}
```

Comment:
Given this function is a no op if code is not '200', is there a reason to produce metrics for error paths? If it is not the case, you may remove the 'code' paramter to this function too.